clang/lib/Driver/Driver.cpp

```diff
@@ -8170,8 +8231,12 @@ InputInfoList Driver::BuildJobsForActionNoCache(
           TargetDeviceOffloadKind == Action::OFK_SYCL) {
         if (UI.DependentOffloadKind == Action::OFK_Host)
           Arch = StringRef();
-        else
-          Arch = UI.DependentBoundArch;
+        else {
+          if (TC->getTriple().isSPIR())
+            Arch = BoundArch;
+          else
+            Arch = UI.DependentBoundArch;
```

Comment:
Sorry I missed that bit previously, This works for CUDA and AMDGCN backend. As you are now using the boundarch as well, I would expect this to work as well. Do you know what is happening ?

Reply:
Good catch - no longer needed after boundarch dependency improvements were made.

Reply:
:+1: